### PR TITLE
[VL] Enable Spark legacy date formatter if spark.sql.legacy.timeParserPolicy is set to 'LEGACY'

### PR DESCRIPTION
**Pull request by NEUpanning**
## What changes were proposed in this pull request?

[Velox#10966](https://github.com/facebookincubator/velox/pull/10966) introduced simple date time formatter that aligns with Spark legacy date parsing/formatting behavior. This PR enables Velox simple date time formatter if `spark.sql.legacy.timeParserPolicy` is set to `LEGACY`.

Relates Velox issue: [#10354](https://github.com/facebookincubator/velox/issues/10354)

## How was this patch tested?

Integration tests with Spark

### CODE REVIEW

**Comment by NEUpanning** [comment]
@rui-mo Could you help to take a look please? Thanks.

File: cpp/velox/compute/WholeStageResultIterator.cc [hunk 1]
```diff
@@ -520,6 +520,11 @@ std::unordered_map<std::string, std::string> WholeStageResultIterator::getQueryC
 
     configs[velox::core::QueryConfig::kSparkPartitionId] = std::to_string(taskInfo_.partitionId);
 
+    // Enable Spark legacy date formatter if spark.sql.legacy.timeParserPolicy is set to 'LEGACY'.
+    if (veloxCfg_->get<std::string>(kSparkLegacyTimeParserPolicy, "") == "LEGACY") {
+      configs[velox::core::QueryConfig::kSparkLegacyDateFormatter] = "true";
```
Comment (rui-mo) [inline cpp/velox/compute/WholeStageResultIterator.cc:525]: Do we need to set this config as false if 'kSparkLegacyTimeParserPolicy' is not 'LEGACY'?

Comment (NEUpanning) [reply cpp/velox/compute/WholeStageResultIterator.cc:525]: Thanks for your feedback. It is necessary to do this. Updated.

**Review by rui-mo — APPROVED** [review]

**Comment by NEUpanning** [comment]
@rui-mo Could you help to merge this PR? Thanks.

**Comment by NEUpanning** [comment]
@surnaik This PR only works for `spark.sql.legacy.timeParserPolicy`, seeing [gluten#7375](https://github.com/apache/incubator-gluten/pull/7375)

**Pull request merged**